sycl/include/syclcompat/launch.hpp

```diff
@@ -118,6 +118,22 @@ launch(const dim3 &grid, const dim3 &threads, Args... args) {
   return launch<F>(grid, threads, get_default_queue(), args...);
 }
 
+// Overload taking zero-argument function object
+template <class F, int Dim>
+std::enable_if_t<std::is_invocable_v<const F &>, sycl::event>
+launch(const F &f, const sycl::nd_range<Dim> &range,
```

Comment:
Unsure what order of argument is best:
- Advantages of lambda first (this): function name (as part of lambda) is first like in cuda and other overloads
- Advantage of lambda last (like in https://github.com/codeplaysoftware/cutlass-fork/pull/305/files): consistent with SYCL parallel_for and sycl_khr_free_function_commands. Function arguments (as part of lambda) are last like in cuda and other overloads.